include/inviwo/core/properties/optionproperty.h

```diff
@@ -108,6 +108,27 @@ class OptionPropertyOption : public Serializable {
     bool operator!=(const OptionPropertyOption<T>& rhs) const;
 };
 
+/**
+ * OptionProperty with a custom type @p T.
```

Comment:
Might mention the requirement for non enum types too. I might also start with that all this is only needed for dynamic properties... which is quite rare...